### ISSUE TITLE
Add builds for Windows on ARM, for Surface Pro X / Surface Pro 9 / etc.

### ISSUE THREAD

**Issue by jmcphers** [rstudio/rstudio#11977]
RStudio is built for Windows only for Intel (x86) processors. However, there is now a version of Windows 11 available for ARM based processors:

https://www.microsoft.com/en-us/software-download/windowsinsiderpreviewARM64

The Surface Pro X from Microsoft is ARM Windows based, and the rumor is that the Surface Pro 9 will be offered in an ARM configuration:
 
https://www.windowscentral.com/hardware/laptops/microsoft-to-merge-surface-pro-x-and-surface-pro-9-under-one-product-line-this-fall

This has lead to a lot of people asking whether they can use R and RStudio on their ARM based Windows devices, e.g.:

https://answers.microsoft.com/en-us/surface/forum/all/install-r-studio-on-my-surface-pro-x-is-it/5aa14f69-e1ce-4d00-89bb-e3ee62e1db89

https://www.reddit.com/r/RStudio/comments/xgk9vc/rrstudio_on_windows_on_arm/

In addition, having an ARM based RStudio release for Windows would be useful to the RStudio development team, who often do development on Apple Silicon machines, where it is necessary to use ARM based virtual machines for Windows development.

**Comment by eitsupi** [comment]
Just a question: there is currently no R installer distributed for arm64 Windows, is there?

Of course, it would be great to have RStudio available on more platforms. Thank you very much.

**Comment by jmcphers** [comment]
Nope, not yet.

**Comment by jmcphers** [comment]
A request here for Windows ARM support, on our community forum: 

https://community.rstudio.com/t/quarto-does-not-work-in-rstudio-on-windows-arm64/155782

**Comment by Sammibulin** [comment]
ARM support on Windows would be fantastic.

**Comment by gtritchie** [comment]
https://blog.r-project.org/2024/04/23/r-on-64-bit-arm-windows/

> R 4.4.0, to be released tomorrow, comes with experimental native support for 64-bit ARM Windows machines (aarch64, arm64). Rtools44 with native support for the platform has been released at the beginning of March.

**Comment by gtritchie** [comment]
https://www.electronjs.org/docs/latest/tutorial/windows-arm

**Comment by andrjohns** [comment]
Adding a +1 that RStudio for windows arm64 would be fantastic! Are there any particular dependencies or build issues that would need to be updated? Happy to contribute where possible

**Comment by gtritchie** [comment]
> Adding a +1 that RStudio for windows arm64 would be fantastic! Are there any particular dependencies or build issues that would need to be updated? Happy to contribute where possible

Mostly it's a matter of priorities and limited time. Even if the dependencies are all in place (and I don't think they are quite there yet, but I haven't spent much time looking into it), spinning up an official build will require us to have cloud-based infrastructure for doing builds on ARM Windows in Docker (not sure this is a thing yet, it's somewhat painful even with Intel Windows), machines for doing our testing, and so on. It's possible we can build on an Intel Windows docker container using cross-compilation techniques, but that's another potential complication that might take significant time to hammer out.

Step one is getting it to build locally on an ARM Windows machine, using ARM C++ compiler, ARM-based Electron, node.js, R, cmake etc. Some tools might be OK being Intel-based if they run successfully via emulation. I want to conduct that experiment, but I don't see myself having the time in the next few months.

**Comment by mbedan** [comment]
RStudio for windows arm64 would be great. I have a new Surface Pro and was able to install R, RStudio, and Quarto without errors. I can run R scripts successfully (and quickly) but rendering Quarto qmd files does not successfully complete. It fails when the knitting process begins and a html is being created. I am not a professional programer/R-user/developer/data scientist but I use R a few times a month a work.

**Comment by ottothecow** [comment]
Given the glowing reviews that the new Surface Laptop and Surface Pro are getting and the inclusion of experimental arm64 support in R4.4.0...I think this needs to happen sooner rather than later.

I've made R+RStudio work on ARM with the x86 emulation layer, but it is far from perfect, has compatibility/version issues, and performs poorly.

MS is making a big push towards ARM here, and those of us who are stuck with windows for various reasons see Windows + ARM devices as a reasonable alternative to a Macbook Air.

**Comment by rathan2794** [comment]
I agree that this has to happen sooner rather than later. I just got the Surface laptop with Windows on ARM. Everything works well.  The device is amazing. Everything runs smoothly. The x86 emulation works but is definitely not efficient.

**Comment by kevinushey** [comment]
Proposing we evaluate this for a future release (probably not next; perhaps the following?)

**Comment by mbalcerzyk** [comment]
I installed R 4.4.1 on Surface Pro X (SQ1 ARM64 processor Windows 11) and R gui works well. I needed to use Rtools to compile from source all packages. All that I used, work well (including Rcmdr). 
When I tried to work previously with R 4.2 in emulation, Rgui worked in 32 bit mode but in 64 bit it opened and closed in a few seconds without reaching the R prompt. 
I would like to use Rstudio - did anybody make it work in such a combination: Rstudio emulation (it is a 32 bit Windows I understand) and R 4.4.1 (or any 4.4.X) for ARM64? I doubt that it will work. 
Can somebody with enough experience compile Rstudio for ARM64 Windows 11 and share an installer? I do not have enough experience in compilation. There is Visual Studio 2022 from Microsoft for ARM64 with C++ workflow. There is Java 17 and Java 11 OpenJDK for ARM64 Windows from Microsoft.
Regards
MarcinB

**Comment by kbodwin** [comment]
Another vote from me for this to be prioritized.  I have students with these laptops, and they aren't able to render their quarto assignments via RStudio. :(

**Comment by mbalcerzyk** [comment]
R 4.4.1++ with Rtools is available for Windows 11 ARM64 native compilation. I work quite well with R in this setup,  albeit I didn't try to install Rstudio in emulsion mode (I.e. with AMD64 Rstudio version).

The only issue is that you have to install packages from source with the command in R
install.packages ("Rcpp", type="source")

The installation is a little bit longer as the compilation is done.

Regards

Marcin Balcerzyk

________________________________

**Comment by ZacWarham** [comment]
Adding another vote for this. Our students will need this come the start of the next academic year and our staff would like time to use in advance as well

**Comment by canesalato** [comment]
adding a vote too. Please

**Comment by gtritchie** [comment]
Note that the issue with rendering Quarto documents on Windows/ARM is tracked separately here: https://github.com/rstudio/rstudio/issues/15277.

**Comment by Visorbearer** [comment]
Any new information on this?

**Comment by remlapmot** [comment]
There is now a Windows 11 on ARM GitHub Actions runner `windows-11-arm`

<https://github.blog/changelog/2025-04-14-windows-arm64-hosted-runners-now-available-in-public-preview/>

**Comment by Visorbearer** [comment]
Thanks! Would this be able to support Quarto, or is that still unavailable?

**Comment by remlapmot** [comment]
Quarto (which is x86) already runs very well on the Windows 11 on Arm (automatically) under emulation. Just install it (that GitHub Actions runner has Chocolatey installed, so it's as simple as in a Terminal: `choco install quarto`) and it runs under the emulation automatically when you run it.

The issue is which version of R that Quarto can run using its knitr engine. Basically it can't run x86 R (which unfortunately is the version RStudio Desktop can run) but it can run AARCH64 R if you run Quarto from the Terminal (and have AARCH64 R on your `PATH` environment variable and AARCH64 RTools45 installed). You can download the AARCH64 versions of R and RTools45 at

<https://www.r-project.org/nosvn/winutf8/aarch64/R-4-signed/R-4.5.1-aarch64.exe>

<https://cran.r-project.org/bin/windows/Rtools/rtools45/files/rtools45-aarch64-6608-6492.exe>

And then put the directory with the `R.exe` and `Rscript.exe` binaries on your `PATH` environment variable; e.g. in Powershell

```powershell
$ENV:PATH = "C:\Program Files\R-aarch64\R-4.5.1\bin;$ENV:PATH"
```

Or if you want to use x86 R you can run that through Quarto by installing Python and Jupyter (I think both x86 and AARCH64 Python will work) and the IRkernel package in R to run the `ir` kernel. Your YAML front matter in your qmd file would be

```yaml
jupyter: ir
```

And I believe the `jupyter: python3` kernel works no problem also (I haven't tested the Julia Jupyter kernel nor the Julia engine yet).

So basically the main obvious thing people want to do - click the _Render_ button in RStudio Desktop on a qmd file using the knitr engine is unfortunately about the only thing that currently doesn't work under emulation. But as soon as RStudio Desktop can run the AARCH64 Windows R then that will work (or indeed if the Windows 11 on Arm emulation is improved to run x86 R trouble free).

**Comment by ShadyGuyJose** [comment]
Hi! How far along is this? I have students who use Snapdragon laptops and they're at a serious disadvantage when compared to their Windows x64 or macOS ARM classmates...

**Comment by srvanderplas** [comment]
I'll echo @ShadyGuyJose here -- I have a student with one of these laptops that is having a hell of a time with R+quarto. I've set up an RStudio server on my home machine to try to minimize the impact on him for the moment, but these computers aren't exactly rare, so there really should be better support.

**Comment by ottothecow** [comment]
> I'll echo [@ShadyGuyJose](https://github.com/ShadyGuyJose) here -- I have a student with one of these laptops that is having a hell of a time with R+quarto. I've set up an RStudio server on my home machine to try to minimize the impact on him for the moment, but these computers aren't exactly rare, so there really should be better support.

I wonder if we can get Microsoft to throw their weight behind making this work.  They are pushing Windows ARM systems hard (manufacturing several themselves) and have historically had a fair amount of involvement in R development.  

It is a real bummer to have to tell students they shouldn't be buying these devices if they plan to take any stats-related classes.

**Comment by ShadyGuyJose** [comment]
> I wonder if we can get Microsoft to throw their weight behind making this work. They are pushing Windows ARM systems hard (manufacturing several themselves) and have historically had a fair amount of involvement in R development.
> 
> It is a real bummer to have to tell students they shouldn't be buying these devices if they plan to take any stats-related classes.

Exactly. And, where I teach, students can't just "plan to take" classes (the university system is radically different from the US). They have a stats class (mine), period, and, while the department provides AiO i7-12700K desktops they can use in class and for exams, they need to be able to practice at home. We never had an issue when everyone had either x86 Windows or x86 Mac (fortunately, we've never had students who couldn't afford a computer at all, and Chromebooks never took off here), and macOS ARM versions came out soon enough, but ARM Windows laptops are becoming reasonably common and there are a few students with those who have issues.

**Comment by aleortiz-uchicago** [comment]
Adding another vote here, specifically from an industry perspective.

Many enterprise environments are beginning to transition their fleets to ARM-based Windows devices (like the Surface Pro and new Snapdragon X Elite laptops) because of the superior battery life and efficiency required for mobile work.

Currently, data scientists and analysts in these firms are hitting a wall. While IT departments are pushing these efficient ARM devices, the lack of a native RStudio build means we can’t utilize the full power of the hardware. Running under emulation causes significant friction with Quarto rendering and package compilation, which are critical for professional reporting. Having native support would prevent RStudio from becoming a 'blocker' for companies looking to modernize their hardware standards.

**Comment by cscheid** [comment]
> Running under emulation causes significant friction with Quarto rendering and package compilatio

(Quarto dev here) Can you describe a bit more what this friction is?

The fundamental constraint is that Quarto is a Deno application, and Deno doesn't currently ship Windows ARM builds. We don't control the development of Deno and so are limited in what we can do in this respect.

**Comment by aleortiz-uchicago** [comment]
Hi @cscheid, thanks for the transparency and for explaining the upstream blocker!

Just a quick heads-up on the Deno constraint it looks like Deno actually just shipped official support for [Windows on ARM with their 2.7 release](https://deno.com/blog/v2.7#windows-on-arm-support).

I know it takes time to test and bump core dependencies, but since the primary upstream blocker might be lifted now, is there any chance the Quarto team might be able to evaluate this for an upcoming roadmap?

To respond to your question about friction, I think the issues are mostly cognitive, though some are technical. For example, people are used to simply clicking the render button. Getting them to switch to a different rendering method requires not only time to teach them, but also convincing. When doing it for one person it is manageable, but scaling that retraining to an entire department becomes unfeasible.

On the technical side, bridging the right version of R using PATH modifications and Jupyter hacks is a fragile process. This means it can easily be set up incorrectly. There is also a performance hit under emulation, which is frustrating because we all just want workflows to run as quickly as possible.

Thanks again for all the work you do!

**Comment by cscheid** [comment]
> To respond to your question about friction, I think the issues are mostly cognitive, though some are technical. For example, people are used to simply clicking the render button. Getting them to switch to a different rendering method requires not only time to teach them, but also convincing. When doing it for one person it is manageable, but scaling that retraining to an entire department becomes unfeasible.

Sorry, but I don't understand what you're saying - ARM emulation shouldn't change the UX at all.

> Just a quick heads-up on the Deno constraint it looks like Deno actually just shipped official support for [Windows on ARM with their 2.7 release](https://deno.com/blog/v2.7#windows-on-arm-support).

@cderv reminds me that Pandoc is also currently lacking a Windows ARM build, and they're blocked on the Haskell compiler.

**Comment by aleortiz-uchicago** [comment]
> Sorry, but I don't understand what you're saying - ARM emulation shouldn't change the UX at all.

You are right, I was talking about the manual steps required when users try to bypass that emulator to get better performance by mixing native ARM tools with emulated Intel tools. Performance is very important. 

> @cderv reminds me that Pandoc is also currently lacking a Windows ARM build, and they're blocked on the Haskell compiler.

For anyone in the future who also wants this you can give a thumbs up for this issue [here](https://gitlab.haskell.org/ghc/ghc/-/issues/25974).

**Comment by gtritchie** [comment]
~~We need to resolve the winpty/conpty issue as part of standing up ARM builds: https://github.com/rstudio/rstudio/issues/3798~~

~~Winpty is abandoned (last changed in 2017) and would likely require work to get it building on ARM. Better to move to the built-in Windows-APIs now that our minimum version is Windows-11 (though most Windows-10 builds will work as well, unless they are extremely out-of-date). It's a big change, though~~

UPDATE: this work has been completed as part of the 2026.06 release.

**Comment by gtritchie** [comment]
Next step is to get an ARM build of rsession included in the Windows package, this will let users use the native ARM R for Windows, even if we don't have the entire app running natively in ARM. This is an intermediate step; if we can get the whole thing done in one release we will, but getting the rsession itself is the top priority.

**Comment by gtritchie** [comment]
Community effort on ARM R for Windows: https://contributor.r-project.org/windows-arm64/.
